[openwrt-23.05] mediatek: enable rootwait for cmcc rax3000m emmc version

1715173329 opened the pull request:
Sometimes the mmc deivce may come up later than kernel attempts to mount rootfs, resulting kernel panic.
Enable rootwait to fix it.

1715173329 committed:
mediatek: enable rootwait for cmcc rax3000m emmc version

Sometimes the mmc deivce may come up later than kernel attempts to
mount rootfs, resulting kernel panic. Enable rootwait to fix it.

Reported-by: Yangyu Chen <cyy@cyyself.name>
Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>
Link: https://github.com/openwrt/openwrt/pull/15077
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

dangowrt commented:
Won't it be easier to add this to cmdline using `/chosen/` section in DTS rather than having to replace the bootloader (again) just for that?

1715173329 commented:
> Won't it be easier to add this to cmdline using `/chosen/` section in DTS rather than having to replace the bootloader (again) just for that?

Yes I added `bootargs-append = " rootwait";` to chosen as well.

hauke commented:
Please add this to the main branch first. After it was merged into main we can backport it to openwrt 3.05

1715173329 commented:
> Please add this to the main branch first. After it was merged into main we can backport it to openwrt 3.05

In main branch it will be fixed in different approach, FYI #15054.

cyyself commented:
There is no action for one month. I think this branch is ready now. @dangowrt

1715173329 committed:
mediatek: fix lan/wan macaddr for cmcc rax3000m

The MAC address assigned to lan/wan was reversed on eMMC boards.

Fixes: d3c193525e62 ("mediatek: add CMCC RAX3000M support")
Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>
Link: https://github.com/openwrt/openwrt/pull/15077
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

1715173329 commented:
kindly ping.

rmilecki commented:
I just checked the commit `mediatek: fix lan/wan macaddr for cmcc rax3000m` as I'm not U-Boot expert.

In general it makes sense - we had a very similar fix in the commit 84ecd7c939b1 ("`mediatek: fix lan/wan macaddr for jdcloud re-cp-03`'). It seems however the same code you're fixing is present in the `master` / `main` branch. Could you first fix it there, please?

I'm also wondering if it would make sense to replace that code with NVMEM cells. We had such change for the JDCloud RE-CP-03 - see commit 0397932fdd3f ("`mediatek: convert eeprom/macaddr to nvmem format for jdcloud re-cp-03`").

1715173329 commented:
> In general it makes sense - we had a very similar fix in the commit [84ecd7c](https://github.com/openwrt/openwrt/commit/84ecd7c939b1ae536d2ad993dbcd8ad496e1a0cb) ("`mediatek: fix lan/wan macaddr for jdcloud re-cp-03`'). It seems however the same code you're fixing is present in the `master` / `main` branch. Could you first fix it there, please?

Yes, it's currently queued in #15054.

> I'm also wondering if it would make sense to replace that code with NVMEM cells. We had such change for the JDCloud RE-CP-03 - see commit [0397932](https://github.com/openwrt/openwrt/commit/0397932fdd3fedabbba095445e5a5a5057a40538) ("`mediatek: convert eeprom/macaddr to nvmem format for jdcloud re-cp-03`").

NVMEM-on-MMC is not supported on 23.05 branch yet.

1715173329 commented:
#15054 has been merged, guessing this is fine to go.
